packages/mail-client/src/mail-client.ts

```diff
@@ -199,8 +156,17 @@ function createTransport(
     port: mailDestination.port
   };
 
-  if (mailDestination.proxyType === 'OnPremise' && socket) {
-    baseOptions.connection = socket;
+  if (mailDestination.proxyType === 'OnPremise') {
+    if (!mailDestination.proxyConfiguration) {
```

Comment:
[pp] Could you move this to a function so we don't have code duplication?

Reply:
I extracted the url build part out. Looks more clean now. Thanks for your suggestion.